### ISSUE TITLE
backup/sync  between instances

### ISSUE THREAD

**Issue by test2a** [frappe/books#413]
Since the current software is supposed to be run on the desktop, i could not find a way for the data to be used in a multi-user environment. 
is that even possible right now? i would want like 5-10 people working on a single company at once, do we open the same DB or have one master and the others send/recieve data from that? how could the software solve conflicts?

**Comment by 18alantom** [comment]
No, as of now there's no way of doing this, you can store the database file on something like Google Drive and then sync it across all devices but this will still cause conflicts.

There is a plan to add multi-user support but it will take some time.

**Comment by solancer** [comment]
It would be great to see a backup/sync option

**Comment by hdell** [comment]
SQL Lite just does not scale when using File System OpLocks... However, client/server database engines (such as PostgreSQL, MySQL, or Oracle) usually support a higher level of concurrency and allow multiple processes to be writing to the same database at the same time. I would suggest you switch to MySQL/MariaDB then mutli-user support should be a lot easier to implement...

**Comment by promexio** [comment]
What about this request? I think it could be very interesting to have this feature! e.g. on the company creation a setup entry where the user needs to select whether he'll work on a "one-man" company on prem (sqlite) or connect to a multi-user environment (Mariadb).

**Comment by 18alantom** [comment]
I will be adding backup in the next few months. Multi-user might take longer.

As for MariaDB and SQLite, yes SQLite doesn't have concurrent writes, but considering that Frappe Books for multi-user environment is bound to have much fewer users (order of 10s) than say a full fledged ERP, SQLite still suffices (it is the fastest db out there).

Each instance i.e. company, get's their own SQLite db. This may seem counter intuitive but the portability of SQLite means the same instance can support local and cloud usage. And even readonly modes for say online report viewing (SQLite can run in the browser).

Anyways, the final decision of how to handle multi-user will be on the basis of some experimentation.

**Comment by promexio** [comment]
Ok, I'd be glad to contribute to the testing of that experimentation. 
For some very small companies, multiuser support is needed as early as there is one person in charge of the accounting / purchase stuff. 
Even though, if it's just reading reports, I guess some acces via dbeaver or an R{} script would do it, but only until the second person is not working actively ...

**Comment by promexio** [comment]
Any news on this?

**Comment by 18alantom** [comment]
@promexio this will take some time, before I begin with this I need to tackle a bunch of other features and issues. Will update here when I begin with it.

**Comment by erik-gross-hardt** [comment]
I built this #695 maybe that's something you might wanna look into.

**Comment by mildred** [comment]
You might be interested in LiteFs https://github.com/superfly/litefs which does exactly that : replication of a SQLite database, although write requests should be forwarded to the master instance.

**Comment by njmulsqb** [comment]
+1 as I am using Frappe books on multiple devices synced using onedrive. It sometimes causes issues and conflicts.

**Comment by girish3** [comment]
@18alantom Is there any update on this? 
We are a small IT consulting company and we are considering Frappe Books ourselves as well for one of our clients and we need multi-user feature or at least a hassle-free workaround.

**Comment by camfsd** [comment]
+1 for this one, especially support for server-based database engines.

It would be great to see support for accessing Books both at home and on-the-go. Remote database support allows multiple devices to access the same instance without the need for manually syncing/replicating two separate instances. I find that this is something many people will find useful.

I understand that this is a feature which will not be considered until v2, however.
